### ISSUE TITLE
[Bug] Crash during opening Tears of the Kingdom in macos

### ISSUE THREAD

**Issue by shoaibsaikat** [Ryujinx/Ryujinx#7191]
### Description of the issue

App crashes opening totk, with previous saved files in macos 

### Reproduction steps

Open Ryujinx
Double click The legend of Zelda Tears of the Kingdom
After compiling shaders app crashes

### Log file

[Ryujinx_1.1.1373_2024-08-10_14-48-57.log](https://github.com/user-attachments/files/16569014/Ryujinx_1.1.1373_2024-08-10_14-48-57.log)
[Ryujinx_1.1.1373_2024-08-10_14-50-37.log](https://github.com/user-attachments/files/16569015/Ryujinx_1.1.1373_2024-08-10_14-50-37.log)
[Ryujinx_1.1.1373_2024-08-10_14-52-15.log](https://github.com/user-attachments/files/16569016/Ryujinx_1.1.1373_2024-08-10_14-52-15.log)

### OS

MacOS 12.7.5

### Ryujinx version

1.1.1373

### Game version

1.2.1

### CPU

Quad-Core Intel Core i7

### GPU

Nvidia GT650M

### RAM

16GB

### List of applied mods

No mod applied

### Additional context?

Faced similar issue in windows too using the same saved files, in both intel and amd PCs

**Comment by thiagodolabsdevops** [comment]
Same behavior is happening within BotW

**Comment by riperiperi** [comment]
I don’t think this GPU will work on macos, there are some missing capabilities we don’t anticipate.

**Comment by shoaibsaikat** [comment]
It's not about gpu I also tested it with my ryzen desktop, same behavior

**Comment by Edgecrusherr160** [comment]
Have you tried purging your shader cache? Sometimes (though it happens far less these days), I have crashes that are resolved by purging my shade cache. 

Does it happen if you remove your saved file and start fresh? It may be a corrupted save, which could explain why it's happening on both platforms with that same saved file. I back my saves every month or so (or when I finish a major achievement).

Which computer are you playing this one that has an Nvidia GT 650M? A 2012-2013 MacBook Pro? I can't imagine that plays well at all. My 2019 MacBook Pro with a Radeon Pro 5300M struggled to play TotK, to the point that I don't bother.
